src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -140,7 +149,77 @@ public override void Clear()
         {
             base.Clear();
             PreloadedTraceData = null;
-            RunEnabled = true;
+            RunSettings.Reset();
+        }
+
+        /// <summary>
+        /// Start periodic evaluation by the given amount of time. If there
+        /// is an on-going periodic evaluation, an exception will be thrown.
+        /// </summary>
+        /// <param name="milliseconds">The desired amount of time between two 
+        /// evaluations in milliseconds.</param>
+        /// 
+        public void StartPeriodicEvaluation(int milliseconds)
+        {
+            if (pulseMaker == null)
+            {
+                pulseMaker = new PulseMaker();
+            }
+
+            pulseMaker.RunStarted += pulseMaker_RunStarted;
+            EvaluationCompleted += pulseMaker.OnRunExpressionCompleted;
+
+            if (pulseMaker.TimerPeriod != 0)
+            {
+                throw new InvalidOperationException(
+                    "Periodic evaluation cannot be started without stopping");
+            }
+
+            pulseMaker.Start(milliseconds);
+        }
+
+        void pulseMaker_RunStarted(object sender, EventArgs e)
+        {
+            var nodesToUpdate = Nodes.Where(n => n.EnablePeriodicUpdate);
+
+            DynamoModel.OnRequestDispatcherBeginInvoke(() =>
+            {
```

Comment:
We should probably pass a `SynchronizationContext`, but we'll leave that be for now.